### PR TITLE
Review of contributor guides

### PR DESCRIPTION
**Pull request by bdmoore1**
- [ ] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?
InfoDev review of contributor guides.

### CODE REVIEW

File: docs/doc_sources/contributor_guides/building.rst [hunk 1]
```diff
@@ -3,18 +3,20 @@
...
 build of DPC++ to build :py:mod:`dpctl`, especially if you want to enable
-CUDA support or try latest features.
+CUDA* support or try the latest features.
 
 Building using oneAPI DPC++
 ---------------------------
 
-Install oneAPI and graphics drivers to the system prior
-to proceeding further.
+prerequisites
```
Comment (bdmoore1) [inline docs/doc_sources/contributor_guides/building.rst:16]: @oleksandr-pavlyk which drivers does a user need? Is there a site we can point them to? And what does it mean to "Install oneAPI"? Should the user install a specific toolkit? Or just the DPC++ compiler?

Comment (oleksandr-pavlyk) [reply docs/doc_sources/contributor_guides/building.rst]: This part needs to be expanded indeed.  The needed drivers would depend on the user's hardware. If the user intends to target Intel GPUs, drivers should be installed following https://dgpu-docs.intel.com/driver/installation.html as appropriate for the hardware configuration. 

If the user intends to target NVidia, or AMD GPU cards, vendor drivers are required, in addition to CodePlay plugins for the installed version of oneAPI DPC++ compiler.   

If the user wants to target CPU device, OpenCL CPU driver comes as a part of oneAPI DPC++ installation, and should be available out of the box. The CPU driver is also packaged in conda, and is automatically made available using conda activation scripts on Linux, and on Windows (in user-mode). If conda is used by a user with elevated privileges (think GitHub Actions CI), additional steps is required of running a PowerShell script available in ``${CONDA_PREFIX}\Scripts\`. Use script `set-intel-ocl-icd-registry.ps1` to set appropriate registry key, and `unset-intel-ocl-icd-registry.ps1` to remove it.

Comment (bdmoore1) [reply docs/doc_sources/contributor_guides/building.rst]: @oleksandr-pavlyk please review my new commit. I added content based on your comments.

File: docs/doc_sources/contributor_guides/building.rst [hunk 2]
```diff
@@ -13,8 +13,34 @@ CUDA support or try latest features.
...
+- To target a CPU, the OpenCL* CPU driver is included as a part of the
+  oneAPI DPC++ Compiler installation. The CPU
+  driver is also packaged in conda, and is automatically made available using
+  conda activation scripts on Linux*, and on Windows* (in user-mode).
+  If conda is used with elevated privileges in Windows (similar to
+  GitHub Actions CI), a PowerShell script must be run:
+
+  .. code-block:: bash
+
+    ${CONDA_PREFIX}\\Scripts.
```
Comment (oleksandr-pavlyk) [inline docs/doc_sources/contributor_guides/building.rst:38]:
```suggestion
    &${CONDA_PREFIX}\\Scripts\script_name.ps1
```

File: docs/doc_sources/contributor_guides/building.rst [hunk 3]
```diff
@@ -13,8 +13,34 @@ CUDA support or try latest features.
...
+  as `CodePlay plugins <https://codeplay.com/solutions/oneapi/plugins/>`_ to
+  enable hardware targeting.
+- To target a CPU, the OpenCL* CPU driver is included as a part of the
+  oneAPI DPC++ Compiler installation. The CPU
+  driver is also packaged in conda, and is automatically made available using
+  conda activation scripts on Linux*, and on Windows* (in user-mode).
+  If conda is used with elevated privileges in Windows (similar to
+  GitHub Actions CI), a PowerShell script must be run:
+
+  .. code-block:: bash
```
Comment (oleksandr-pavlyk) [inline docs/doc_sources/contributor_guides/building.rst:36]:
```suggestion
  .. code-block:: powershell
```

File: docs/doc_sources/contributor_guides/building.rst [hunk 4]
```diff
@@ -13,8 +13,34 @@ CUDA support or try latest features.
...
+- To target a CPU, the OpenCL* CPU driver is included as a part of the
+  oneAPI DPC++ Compiler installation. The CPU
+  driver is also packaged in conda, and is automatically made available using
+  conda activation scripts on Linux*, and on Windows* (in user-mode).
+  If conda is used with elevated privileges in Windows (similar to
+  GitHub Actions CI), a PowerShell script must be run:
+
+  .. code-block:: powershell
+
+    &${CONDA_PREFIX}\\Scripts\script_name.ps1
```
Comment (oleksandr-pavlyk) [inline docs/doc_sources/contributor_guides/building.rst:38]:
```suggestion
    &$Env:CONDA_PREFIX\Scripts\script_name.ps1
```

**Review by oleksandr-pavlyk — APPROVED** [review]

**Review by ndgrigorian — APPROVED** [review]

**Pull request merged**